net/dnscache/dnscache.go

```diff
@@ -147,34 +149,30 @@ var debug = envknob.Bool("TS_DEBUG_DNS_CACHE")
 //
 // If err is nil, ip will be non-nil. The v6 address may be nil even
 // with a nil error.
-func (r *Resolver) LookupIP(ctx context.Context, host string) (ip, v6 net.IP, allIPs []net.IPAddr, err error) {
+func (r *Resolver) LookupIP(ctx context.Context, host string) (ip, v6 netip.Addr, allIPs []netip.Addr, err error) {
 	if r.SingleHostStaticResult != nil {
 		if r.SingleHost != host {
-			return nil, nil, nil, fmt.Errorf("dnscache: unexpected hostname %q doesn't match expected %q", host, r.SingleHost)
+			return zaddr, zaddr, nil, fmt.Errorf("dnscache: unexpected hostname %q doesn't match expected %q", host, r.SingleHost)
 		}
 		for _, naIP := range r.SingleHostStaticResult {
-			ipa := &net.IPAddr{
-				IP:   naIP.AsSlice(),
-				Zone: naIP.Zone(),
-			}
-			if ip == nil && naIP.Is4() {
-				ip = ipa.IP
+			if !ip.IsValid() && naIP.Is4() {
+				ip = naIP
 			}
-			if v6 == nil && naIP.Is6() {
-				v6 = ipa.IP
+			if !v6.IsValid() && naIP.Is6() {
+				v6 = naIP
 			}
-			allIPs = append(allIPs, *ipa)
+			allIPs = append(allIPs, naIP)
 		}
 		return
 	}
 	if ip := net.ParseIP(host); ip != nil {
```

Comment:
use netip.ParseAddr instead?

Reply:
then you don't even need ip2addr?

Reply:
Oh, I don't think I even noticed that when I first wrote this code. Updated.